packages/vuetify/src/labs/VMaskInput/__tests__/VMaskInput.spec.browser.tsx

```diff
@@ -31,4 +31,319 @@ describe('VMaskInput', () => {
+
+        await insertCaretAt(10, 12)
+        await userEvent.keyboard('{Ctrl>}x{/Ctrl}')
+
+        expect(input.selectionStart).toBe(10)
+        expect(model.value).toBe('(AS)-123-X45-67-')
+      })
+
+      it('selection contains one character before, one at and one after delimiter', async () => {
+        const { input, model, insertCaretAt } = renderComponent()
+
+        await insertCaretAt(11, 14)
+
+        await userEvent.keyboard('{Ctrl>}x{/Ctrl}')
+        expect(input.selectionStart).toBe(11)
+        expect(model.value).toBe('(AS)-123-XY5-67-')
+      })
+    })
+
+    describe('Paste', () => {
+      it('pasted when cursor is at end', async () => {
+        const { input, model, insertCaretAt } = renderComponent({ defaultMask: '####', defaultModel: '12' })
+
+        navigator.clipboard.writeText('34')
+
+        await insertCaretAt(2)
+        await userEvent.paste()
+
+        expect(input.selectionStart).toBe(4)
+        expect(model.value).toBe('1234')
+      })
+
+      it('pasted when cursor is in middle', async () => {
+        const { input, model, insertCaretAt } = renderComponent({ defaultMask: '#####', defaultModel: '1245' })
+
+        navigator.clipboard.writeText('3')
+        await insertCaretAt(2)
+        await userEvent.paste()
+
+        expect(input.selectionStart).toBe(3)
+        expect(model.value).toBe('12345')
+      })
+
+      it('pasted when cursor is after delimiter', async () => {
+        const { input, model, insertCaretAt } = renderComponent({ defaultModel: '(AS)-123-XYZ-' })
+
+        navigator.clipboard.writeText('12')
+        await insertCaretAt(13)
+        await userEvent.paste()
+
+        expect(input.selectionStart).toBe(16)
+        expect(model.value).toBe('(AS)-123-XYZ-12-')
+      })
+
+      it('pasted when cursor is before delimiter', async () => {
+        const { input, model, insertCaretAt } = renderComponent({ defaultMask: '##-##', defaultModel: '12-' })
+
+        navigator.clipboard.writeText('34')
+        await insertCaretAt(2)
+        await userEvent.paste()
+
+        expect(input.selectionStart).toBe(5)
+        expect(model.value).toBe('12-34')
+      })
+
+      it('pasted when selection is in middle', async () => {
+        const { input, model, insertCaretAt } = renderComponent()
+
+        navigator.clipboard.writeText('CD')
+        await insertCaretAt(10, 12)
+        await userEvent.paste()
+
+        // TODO: Fix this test
+        expect(input.selectionStart).toBe(12)
+        expect(model.value).toBe('(AS)-123-XCD-45-67')
+      })
+
+      it('pasted when selection is in middle', async () => {
+        const { input, model, insertCaretAt } = renderComponent()
+
+        navigator.clipboard.writeText('CD')
+        await insertCaretAt(1, 3)
+        await userEvent.paste()
+
+        // TODO: Fix this test
+        expect(input.selectionStart).toBe(5)
+        expect(model.value).toBe('(CD)-123-XCD-45-67')
```

Comment:
I'd flip the order, first assert the value, then the caret position. It was easier to work with the pasting handler when I could make sure the value is right and only then focus on the caret, which is arguably a secondary concern.